### PR TITLE
Blog app: add API endpoints 🎯

### PR DESCRIPTION
**Pull request by ritikarawat220**
### Project requirements  💭

- Create an API endpoint to list all posts for a user.
- Create an API endpoint to list all comments for a user's post.
- Create an API endpoint to add a comment to a post. Remember that the owner of the comment is the user that makes it; it should not be set by passing a parameter.
- Your API endpoints should receive JSON and respond JSON as well.

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -1,19 +1,43 @@
...
+  # namespace :api do
+  #   namespace :v1 do
+  #     get 'posts/index'
+  #   end
+  # end
+  # namespace :api do
+  #   namespace :v1 do
+  #     get 'users/index'
+  #   end
+  # end
```
Comment (tresorsawasawa) [inline config/routes.rb]: - [ ] For best practices and clean code, kindly ensure that you remove all the commented and unused code lines from your code base

**Review by tresorsawasawa — CHANGES_REQUESTED** [review]
# STATUS: REQUIRED CHANGES :recycle::recycle: 

Hi @ritikarawat220, @MMhaidari, and @jeanbuhendwa 
Good job so far! I know you worked hard to implement this project, I really appreciate the effort you put in! But there are some issues that you still need to work on to go to the next project but you are almost there! 

almost-there

## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Good PR description :heavy_check_mark: 
- Follow Gitflow :heavy_check_mark: 

## Required changes :recycle: :recycle: 
- [ ] Check the comments under the review.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear. 
Please ping me [@tresorsawasawa](https://github.com/tresorsawasawa) when you comment so I can receive the notification or use slack with the same name._

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: config/routes.rb [hunk 2]
```diff
@@ -1,19 +1,32 @@
...
+
   # devise_for :users
-  devise_for :users, controllers: { passwords: 'devise/passwords' }
+  devise_for :users, controllers: { sessions: 'users/sessions', passwords: 'devise/passwords' }
   
   # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
 
   # Defines the root path route ("/")
   # root "articles#index"
   # root to: 'home#index'
```
Comment (Felix45) [inline config/routes.rb]: - [ Required ] Good job deleting some of the commented code lines from your project as requested by the previous reviewer, however, this file still has commented code. Please consider deleting the commented pieces of code from your codebase to make your code clean and professional.

File: config/routes.rb [hunk 3]
```diff
@@ -1,19 +1,32 @@
 Rails.application.routes.draw do
+  get 'sessions/new'
+  get 'sessions/create'
+
   # devise_for :users
```
Comment (Felix45) [inline config/routes.rb]: - [ Required ] Good job deleting some of the commented code lines from your project as requested by the previous reviewer, however, this file still has commented code. Please consider deleting the commented pieces of code from your codebase to make your code clean and professional.

**Review by Felix45 — CHANGES_REQUESTED** [review]
Hi @ritikarawat220, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 
## Highlights :heavy_check_mark: 
- [x] Git flow is used correctly :white_check_mark: 
- [x] Pull request has a descriptive summary and title :white_check_mark: 
- [x] Repository has a professional readme file :white_check_mark:  
- [x] Implemented some of the changes from the previous reviewer :white_check_mark:   


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by danmainah — APPROVED** [review]
Hi @Team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🎉 🎉 

## Highlights
✔️  Implemented the requested to ensure the codebase is clean and professional by removing all commented code.
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**